src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/admin/AdminCreateSessionsTemplateTest.kt

```diff
@@ -72,4 +77,239 @@ class AdminCreateSessionsTemplateTest : IntegrationTestBase() {
     Assertions.assertThat(sessionTemplateDto.prisonerIncentiveLevelGroups.stream().map { it.incentiveLevels }).containsExactlyInAnyOrder(enhancedIncentives)
     Assertions.assertThat(sessionTemplateDto.prisonerIncentiveLevelGroups[0].reference).isEqualTo(dto.incentiveLevelGroupReferences!![0])
   }
+
+  @Test
+  fun `when session template name greater than 100 then validation fails and BAD_REQUEST is returned`() {
+    // Given
+    val dto = createSessionTemplateDto(
+      name = RandomStringUtils.randomAlphabetic(101),
+    )
+
+    // When
+    val responseSpec = callCreateSessionTemplate(webTestClient, dto, setAuthorisation(roles = adminRole))
+
+    // Then
+    responseSpec.expectStatus().isBadRequest
```

Comment:
Would be good if you can test to see if the message is correct,... Might be good to add to VisitSchedulerExceptionHandler a method to convert the javax message into the ErrorResponse format.